### ISSUE TITLE
Bug: [new config system] --ignore-path option disappears when eslint.config.js is used (is this undocumented behavior?)

### ISSUE THREAD

**Issue by jedwards1211** [eslint/eslint#16433]
### Environment

Node version: v16.15.1
npm version: v8.11.0
Local ESLint version: v8.25.0 (Currently used)
Global ESLint version: Not found
Operating System: darwin 21.6.0

### What parser are you using?

Default (Espree)

### What did you do?

```sh
> mkdir eslint-bug
> cd eslint-bug
> npm init
> npm i eslint
> touch .gitignore
> eslint --ignore-path .gitignore # succeeds
> touch eslint.config.js
> eslint --ignore-path .gitignore
Invalid option '--ignore-path' - perhaps you meant '--ignore-pattern'?
```

### What did you expect to happen?

`--ignore-path` works in any case.  Is `--ignore-path` intentionally not supported when using `eslint.config.js`?  If so then [the docs for `--ignore-path`](https://eslint.org/docs/latest/user-guide/command-line-interface#--ignore-path) need to make this clear.

### What actually happened?

When `eslint.config.js` exists, running `eslint --ignore-path .gitignore` errors out: `Invalid option '--ignore-path' - perhaps you meant '--ignore-pattern'`

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

The option disappears from `eslint -h` as well.

**Comment by mdjermanovic** [comment]
Hi @jedwards1211, thanks for the issue!

> Is `--ignore-path` intentionally not supported when using `eslint.config.js`? If so then [the docs for `--ignore-path`](https://eslint.org/docs/latest/user-guide/command-line-interface#--ignore-path) need to make this clear.

That's correct, `--ignore-path` and a few other CLI options are intentionally not supported when using `eslint.config.js`. I agree that it would be good to document this, but I'm not sure if it would be better to update the CLI docs or new config files docs. Since there are other features that are not supported when using `eslint.config.js`, such as automatically loading `.eslintignore` files and `/* env */` directives, it might be better to add a new section to [Configuration Files (New)](https://eslint.org/docs/latest/user-guide/configuring/configuration-files-new) listing all the unsupported features. @nzakas what do you think?

**Comment by nzakas** [comment]
Sure, we can add a "Differences from eslintrc" section in the flat config docs.

I don't want to start changing other places in the docs until we have finalized the features and behavior of flat config to keep the maintenance burden as low as possible.

**Comment by jedwards1211** [comment]
If code is released to production, then to me, the more relevant places the behavior is documented, the better

**Comment by jedwards1211** [comment]
@nzakas I think it would be good to note in the warning at the top of the new config files docs that (AFAICT) the flat config features aren't guaranteed to adhere to semver compatibility between minor and patch releases.
